lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexFuture.java

```diff
@@ -32,7 +34,8 @@
  * The interface between user code and submitted task.
  */
 @Unstable
-public final class VortexFuture<TOutput> implements Future<TOutput> {
+public final class VortexFuture<TOutput extends Serializable>
+    implements Future<TOutput>, TaskletStateDelegate<TOutput> {
```

Comment:
Should we remove ``implements Future<TOutput>``? I'm concerned that users will use it with other libraries that introduces scenarios that we don't support or haven't thought of supporting.